src/pe/header.rs

```diff
@@ -393,17 +409,21 @@ impl DosHeader {
 ///
 /// * De facto, can be followed by a non-standard ["Rich header"](https://0xrick.github.io/win-internals/pe3/#rich-header).
 /// * According to the standard, is followed by the  [Header::signature] and then the [CoffHeader].
-pub struct DosStub(pub [u8; 0x40]);
-impl Default for DosStub {
+pub struct DosStub<'a> {
+    pub data: &'a [u8],
```

Comment:
this is a breaking change, going from newtype struct to regular struct with named field

Reply:
Unfortunately, there is no choice but to make this breaking change. Since the dos stub never be fixed size. It is very specific to the toolchain and compilers.

For ex:

- MSVC(link.exe): 64-bytes
- Clang(LLD): 54-bytes[^1][^2]
- Borland: 64-bytes, with unique signatures: `MZP` and "This program must be run under Win32"

So technically, it is implemented in wrong way to make it `Sized`. I believe we cannot avoid breaking changes to fix this problem and make goblin more stabilize and reliable in that way.

[^1]: https://github.com/llvm/llvm-project/blob/54d3cf14213f18e44ef9ed2ffe3b3131e472e2f5/lld/COFF/Writer.cpp#L70C1-L76C3
[^2]: This makes the `MZx` while is `MZ.` on MSVC, so I usually identifies the binary by this `MZx` to whether or it is compiled with MSVC linker. If `MZP`, it's Borland.

Reply:
So... maybe, ways if we would make DOS stub dynamically sized, I think of:

- **Breaking Change**: To have explicit lifetimes (as well as I did ealier; reverted recently) and do not copy any (regardless of trivial or big) buffers (`&'a [u8]`s).
  - Pros: Great code, no deep copy.
  - Cons: Codebase would potentially go to the hell of "lifetimes" and coding will not be easy for beginner contributors
- **Breaking Change**: Keep implicit lifetimes. Remove `Copy` impls from appropriate structs around `Header`, and abuse `clone`s, deep copy, and deal with compatibility problems within the current codebase.
  - Pros: The effect is minimum.
  - Cons: with performance tradeoffs.

Reply:
Yeah.. something to related, I found another issue. During implementing tests I noticed that the DOS header produced by Borland(Indeed Turbo incremental linker included in Borland C++ Compiler) is not usual in any way and we need more changes to make goblin stable.

For short: The fixed `PE_POINTER_OFFSET` should be removed and must be dynamizally calculated. PE pointer offset of 
the binary in question is indeed `0x200`.

|Borland|MSVC|
|-|-|
|![borland](https://github.com/user-attachments/assets/47c6c156-527c-4494-9819-3298b4e147c4)|![msvc](https://github.com/user-attachments/assets/ec9feb68-9637-4d41-9517-c8ad86621c39)|

Reply:
Fixed in 870cad9ab5d81778f9845f88f5a3b9b12e14741c.

Alright, breaking change again but it's inevitable. Removed `const PE_POINTER_OFFSET`, which can lead to the bug code and codes referenced that have been adjusted with correct calculation of PE pointer offset.

I removed an one assert that asserts that DOS header is ends at `PE_POINTER_OFFSET`, which is wrong implementation, as I mentioned ealier. No alternative assertion is required, since DOS stub bytes can be anything that is a valid DOS program bytecodes in theory.

With those changes, to ensure it is still correct and improved, I added the tests for Borland headers, and is now everything is passed.

Addiotonally, I added the detailed explaination on the default DOS stub byte code (in `impl Default for DosStub`). I hope that improves docs as well.

I'd like to hear from you about how to handle those inevitable breaking changes at this time.